homeassistant/components/overkiz/number.py

```diff
@@ -58,26 +63,32 @@ class OverkizNumberDescription(NumberEntityDescription, OverkizNumberDescription
         name="Eco Room Temperature",
         icon="mdi:thermometer",
         command=OverkizCommand.SET_ECO_TEMPERATURE,
-        min_value=6,
-        max_value=29,
+        device_class=NumberDeviceClass.TEMPERATURE,
```

Comment:
This PR sets a device class, but also an icon. 
We should not set an icon when the device class is set.

Reply:
May I ask why? We shouldn’t override the one set by default for the device class?

Reply:
The device class provides the icon for the frontend (or at least it should) when using a device class. In this case, the temperature device class should already provide the default `mdi:thermometer` icon.

Setting it here, will add it to memory/state machine, store it in the database and sends it over the wire as well; while it is simply not needed.

Reply:
> Does frontend know about number device classes @frenck ? 

It should, it is part of adding a new device class to HA.
Core, end-user docs, dev docs, frontend support, and (added bonus) inform the ESPHome project so it can be handled with them upstream too.